src/main/scala/sbtghactions/WorkflowStep.scala

```diff
@@ -33,17 +33,31 @@ object WorkflowStep {
   val DefaultSbtStepPreamble: List[String] = List(s"++ $${{ matrix.scala }}")
 
   val CheckoutFull: WorkflowStep = Use(
-    UseRef.Public("actions", "checkout", "v5"),
+    UseRef.Public(
+      "actions",
+      "checkout",
+      "08c6903cd8c0fde910a37f88322edcfb5dd907a8",
+      "v5.0.0"),
```

Comment:
I'm sympathetic to having an immutable and repeatable CI configuration, but the downside to embedding git sha into `*.scala` code this way is that now the upgrade burden would shift to the sbt-github-actions maintainer and/or simply this would be left forgotten as the actions might ship fixes and patches.

Reply:
Jup - this PR was more of a 'straw man' to get the conversation going than something I'd expect to get merged in this form.

What would be the way forward here? Perhaps we could have an option that makes `githubWorkflowCheck` more lenient and allow dependabot to update versions (but not the rest of the workflow)?

Reply:
Perhaps we could keep using tags/branches but 'accept' the hashes in the check, like #207 ?